suite/tests/linux/syscall_pwait.c

```diff
@@ -93,38 +150,43 @@ main(int argc, char *argv[])
 
     INIT();
 
+    ready_to_listen = mmap(NULL, sizeof(*ready_to_listen), PROT_READ | PROT_WRITE,
+                           MAP_SHARED | MAP_ANONYMOUS, -1, 0);
+
     sleeptime.tv_sec = 0;
     sleeptime.tv_nsec = 500 * 1000 * 1000;
 
     intercept_signal(SIGUSR1, (handler_3_t)signal_handler, true);
     intercept_signal(SIGUSR2, (handler_3_t)signal_handler, true);
     print("handlers for signals: %d, %d\n", SIGUSR1, SIGUSR2);
     sigemptyset(&new_set);
+    sigaddset(&new_set, SIGUSR2);
```

Comment:
I think this one line was the only thing that needed to change in the test: without any synch it would have worked, though it is nicer to be deterministic and ensure we test pending signals.

Reply:
Right, but now we found the pending signal issue which was a real bug.. so it's ok.